modules/experimental-layers/src/bitmap-layer/bitmap-layer.js

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {Layer} from '@deck.gl/core';
+import {GL, Model, Geometry, loadTextures} from 'luma.gl';
+
+import BITMAP_VERTEX_SHADER from './bitmap-layer-vertex';
+import BITMAP_FRAGMENT_SHADER from './bitmap-layer-fragment';
+
+// Note: needs to match vertex shader
+const MAX_BITMAPS = 11;
+
+const defaultProps = {
+  images: [],
+
+  desaturate: 0,
+  blendMode: null,
+  // More context: because of the blending mode we're using for ground imagery,
+  // alpha is not effective when blending the bitmap layers with the base map.
+  // Instead we need to manually dim/blend rgb values with a background color.
+  transparentColor: [0, 0, 0, 0],
+  tintColor: [255, 255, 255],
+  // accessors
+  getCenter: x => x.center,
+  getRotation: x => x.rotation
+};
+
+/*
+ * @class
+ * @param {object} props
+ * @param {number} props.transparentColor - color to interpret transparency to
+ * @param {number} props.tintColor - color bias
+ */
+export default class BitmapLayer extends Layer {
+  initializeState() {
+    const {gl} = this.context;
+    this.setState({model: this.getModel(gl)});
+
+    const {attributeManager} = this.state;
+    attributeManager.addInstanced({
+      instanceCenter: {size: 3, update: this.calculateInstanceCenters},
+      instanceRotation: {size: 3, update: this.calculateInstanceRotations},
+      instanceBitmapIndex: {size: 1, update: this.calculateInstanceBitmapIndex}
+    });
+  }
+
+  updateState({props, oldProps}) {
+    if (props.images !== oldProps.images) {
+      let changed = !oldProps.images || props.images.length !== oldProps.images.length;
+      if (!changed) {
+        for (let i = 0; i < props.images.length; ++i) {
+          changed = changed || props.images[i] !== oldProps.images[i];
+        }
+      }
+      if (changed) {
```

Comment:
If anything changes, we reload them all? Maybe add some comment about this limitation?